src/shared/condition.c

```diff
@@ -1030,6 +1031,7 @@ static int condition_test_psi(Condition *c, char **env) {
         /* If only one parameter is passed, then we look at the global system pressure rather than a specific cgroup. */
         if (r == 1) {
                 pressure_path = path_join("/proc/pressure", pressure_type);
+                system_wide = true;
                 if (!pressure_path)
```

Comment:
```suggestion
                system_wide = true;

                pressure_path = path_join("/proc/pressure", pressure_type);
                if (!pressure_path)
```

Reply:
yes, please keep the allocation and its immediate error handling together. And assign the boolean either before both or after both, but not in between it